src/tgen-markovmodel.c

```diff
@@ -240,7 +240,8 @@ static gboolean _tgenmarkovmodel_findVertexAttributeString(TGenMarkovModel* mmod
 
     if(igraph_cattribute_has_attr(mmodel->graph, IGRAPH_ATTRIBUTE_VERTEX, name)) {
         igraph_attribute_type_t type = IGRAPH_ATTRIBUTE_DEFAULT;
-        igraph_i_attribute_gettype(mmodel->graph, &type, IGRAPH_ATTRIBUTE_VERTEX, name);
+        g_assert(igraph_cattribute_table.gettype);
+        igraph_cattribute_table.gettype(mmodel->graph, &type, IGRAPH_ATTRIBUTE_VERTEX, name);
```

Comment:
This requires that all versions of igraph expose the `igraph_cattribute_table` and its `gettype` function. Are we confident that is always the case?

Would it be more robust to check that both `igraph_cattribute_table` and `igraph_cattribute_table.gettype` are not NULL, and then return FALSE if either of those are NULL (and maybe log a warning)?

Reply:
> This requires that all versions of igraph expose the igraph_cattribute_table and its gettype function. Are we confident that is always the case?

Well, it works for all the platforms in the CI, meaning `ubuntu-latest` ;)

I suppose tgen is meant to support the same set of platforms that shadow does? Maybe I should add those to the CI in a separate PR first?

> Would it be more robust to check that both igraph_cattribute_table and igraph_cattribute_table.gettype are not NULL, and then return FALSE if either of those are NULL (and maybe log a warning)?

On further inspection I agree that the `g_assert` is insufficient, since it's not clear what guarantees  there are about this table being initialized, it needs to be checked in non-debug builds too.

I'd lean towards exiting with an error rather than just logging a warning, unless you think tgen would still often/usually end up doing the right thing somehow...?

Reply:
Now tested on all shadow-supported platforms. I ended up *removing* the type checks - see updated PR description for reasoning.

Reply:
> The type not being as expected would only happen due to a programming error in tgen.

I don't follow this. Markov model graph files can be configured by the user, so won't they crash tgen if they pass in a Markov model graph with an incorrect type? I thought the code you deleted checked the type and failed gracefully if it found the wrong type. What am I missing?

Reply:
For example, we check attribute types in `_tgenmarkovmodel_checkEdgeAttributeParamExists`, which eventually calls `_tgenmarkovmodel_findEdgeAttributeDouble` which then calls `igraph_i_attribute_gettype `. If the type is incorrect, a warning is printed in `_tgenmarkovmodel_checkEdgeAttributeParamExists`.

Sure, tgen can't continue and will exit, but at least there is a useful log warning message that provides some context.

Reply:
> Markov model graph files can be configured by the user

Ah ok. Yeah I'd misunderstood.

I introduced a type error in one of the test models and checked the behavior. The `main` current behavior  does end up exiting tgen with an error about the missing attribute.

Also it turns out I was wrong about the behavior when fetching an attribute using a mismatched type. In particular when fetching a string as a double, I'd expected igraph to abort (which wouldn't be ideally user friendly, but not too terrible), or return NaN (in which case we'd basically get the old behavior of logged warning and graceful exit), but instead it actually returns 0.0. Yikes.

Added a helper function to use `igraph_cattribute_table.gettype` when available to preserve the old behavior. It doesn't really look like this function is intended to be called by users of the library, so might not be present at runtime or might break the build again in the future, but I agree now that it's the least bad option :).